resources/views/components/cart-deductions.blade.php

```diff
@@ -0,0 +1,11 @@
+<div {{ $attributes }}>
+
+    <ul>
+        @foreach ($deductions as $deduction)
+            <li>
+                <x-dynamic-component :component="$deduction->getViewComponent() ?? 'tipoff-cart-deduction'" :deduction="$deduction"/>
```

Comment:
This is a view component that lists all of the deductions.  The elements in the list use `x-dynamic-component` that allows swapping to deduction specific component, but with a fallback to a generic component when none is supplied.